### PR TITLE
[Markdown] clickable urls

### LINKED ISSUE #80: [markdown] recognize URLs and create links

### PR DESCRIPTION
**Pull request by DavidBadura**
fix #80

### CODE REVIEW

File: src/AppBundle/Markdown/UrlParser.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+use League\CommonMark\Inline\Element\Link;
+use League\CommonMark\Inline\Parser\AbstractInlineParser;
+use League\CommonMark\InlineParserContext;
+
+/**
+ * @author David Badura <d.a.badura@gmail.com>
+ */
+class UrlParser extends AbstractInlineParser
+{
+    const PATTERN = '#(?i)\b((?:[a-z][\w-]+:(?:/{1,3}|[a-z0-9%])|www\d{0,3}[.]|[a-z0-9.\-]+[.][a-z]{2,4}/)(?:[^\s()<>]+|\(([^\s()<>]+|(\([^\s()<>]+\)))*\))+(?:\(([^\s()<>]+|(\([^\s()<>]+\)))*\)|[^\s`!()\[\]{};:\'".,<>?«»“”‘’]))#iS';
```
Comment (tolry) [inline src/AppBundle/Markdown/UrlParser.php]: @DavidBadura that's easy to understand and comprehensive :wink: I guess you didn't write that regular expression yourself? Would be an excellent point to introduce some unit tests, as those tests would illustrate which URLs are detected and which are not, e.g. is file:///... supported? etc.

File: src/AppBundle/Markdown/UrlParser.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    /**
+     * @return string[]
+     */
+    public function getCharacters()
```
Comment (tolry) [inline src/AppBundle/Markdown/UrlParser.php]: @DavidBadura  What does this method do? I might read the League\CommonMark config, but a short description in the comments might help to speed up understanding. I would guess, this means, this extension is only triggered, when encountering an 'h' or 'f' character? Which would make a www.some-domain.com link not working?

File: src/AppBundle/Markdown/UrlParser.php (hunk 1 above)
Comment (DavidBadura) [inline src/AppBundle/Markdown/UrlParser.php]: i will try to add a test :D

File: src/AppBundle/Markdown/UrlParser.php (hunk 2 above)
Comment (DavidBadura) [inline src/AppBundle/Markdown/UrlParser.php]: yes, if i add the character `w` into the list, then www.some-domain.com will be detected. but to be a absolute path i need to prefix it with `http://` or `//`

File: src/AppBundle/Markdown/UrlParser.php (hunk 1 above)
Comment (DavidBadura) [inline src/AppBundle/Markdown/UrlParser.php]: haha, this project has any tests :D

File: src/AppBundle/Markdown/UrlParser.php (hunk 2 above)
Comment (DavidBadura) [inline src/AppBundle/Markdown/UrlParser.php]: www.some-domain.com works now

File: src/AppBundle/Markdown/UrlParser.php (hunk 1 above)
Comment (DavidBadura) [inline src/AppBundle/Markdown/UrlParser.php]: done

**Comment by DavidBadura** [comment]
it's ready

**Pull request merged**